docs/architecture/widget_framework.md

```diff
@@ -0,0 +1,90 @@
+# Widget framework
+
+The widget framework comprises common code that provides a "backbone structure" to our business logic.
+
+Each screen's content is composed entirely of widgets, and when a screen requests new data, no matter which type it is, the same backend logic is used to look up the screen's configuration, fetch the appropriate real-time data, transform that data into "candidate" widgets, and then assemble some of those candidates together to be sent back to the client and rendered.
+
+## What's a widget?
```

Comment:
~~I know this doc primarily focuses on visuals, but we do have widgets that are audio only. Would that be worth mentioning here?~~

Reply:
I see the section for this at the bottom here, so disregard.